packages/react-reconciler/src/__tests__/ReactLazy-test.internal.js

```diff
@@ -1504,72 +1452,155 @@ describe('ReactLazy', () => {
     expect(root).toMatchRenderedOutput('ba');
   });
 
-  it('mount and reorder lazy elements (legacy mode)', async () => {
-    class Child extends React.Component {
-      componentDidMount() {
-        Scheduler.log('Did mount: ' + this.props.label);
-      }
-      componentDidUpdate() {
-        Scheduler.log('Did update: ' + this.props.label);
+  describe('legacy mode', () => {
+    beforeEach(() => {
+      ReactFeatureFlags.disableLegacyMode = false;
+    });
+
+    afterEach(() => {
+      ReactFeatureFlags.disableLegacyMode = true;
+    });
+
+    it('mount and reorder lazy elements (legacy mode)', async () => {
```

Comment:
Does it work to gate this code instead of setting the feature flag?

```suggestion
    // @gate disableLegacyMode
    it('mount and reorder lazy elements (legacy mode)', async () => {
```

Reply:
They won't fail with the flag off until https://github.com/facebook/react/pull/28498

I've been trying to pull out the bulk of the test migrations out of that PR to simplify things

Reply:
I'll remove the flag setting here and add the gate in https://github.com/facebook/react/pull/28498